File: solutions/2026-01-12/solution.java
// Validate IP Address
// Difficulty: Medium
// URL: https://leetcode.com/problems/validate-ip-address/

public boolean validIPAddress(String IP) {
  String[] parts = IP.split(".");
  if (parts.length != 4) return false;
  for (String part : parts) {
    if (part.length() > 1 && part.charAt(0) == '0') return false;
    for (char c : part.toCharArray()) {
      if (!Character.isDigit(c)) return false;
    }
    int num = Integer.parseInt(part);
    if (num > 255 || num < 0) return false;
  }
  return true;
}

File: solutions/2026-01-12/solution.py
# Validate IP Address
# Difficulty: Medium
# URL: https://leetcode.com/problems/validate-ip-address/

def validIPAddress(IP: str) -> bool:
  parts = IP.split('.')
  if len(parts) != 4:
    return False
  for part in parts:
    if not part.isdigit() or (len(part) > 1 and part[0] == '0'):
      return False
    if int(part) > 255 or int(part) < 0:
      return False
  return True